frontend/components/Chat.tsx

```diff
@@ -0,0 +1,247 @@
+import { useState, useEffect, useRef, useId } from "react";
+import clsx from "clsx";
+import Link from "next/link";
+import { Icon } from "@iconify/react";
+import { client } from "lib/client";
+import { pagesPath } from "lib/$path";
+import useChats from "lib/use-chats";
+
+function Wait() {
+  return (
+    <svg
+      className="fill-gray-500"
+      viewBox="0 0 40 12"
+      width="40"
+      height="16"
+      role="img"
+    >
+      <circle className="animate-bounce" cx="4" cy="8" r="4" />
+      <circle
+        className="animate-bounce [animation-delay:0.3s]"
+        cx="20"
+        cy="8"
+        r="4"
+      />
+      <circle
+        className="animate-bounce [animation-delay:0.6s]"
+        cx="36"
+        cy="8"
+        r="4"
+      />
+    </svg>
+  );
+}
+
+const sleep = (timeout: number) =>
+  new Promise<void>((resolve) => {
+    setTimeout(() => {
+      resolve();
+    }, timeout);
+  });
+
+type Props = {
+  className?: string;
+};
+
+function Chat({ className }: Props) {
+  const { chats, pushChat, popChat } = useChats();
+  const chatsRef = useRef<HTMLUListElement>(null);
+  useEffect(() => {
+    if (!chatsRef.current) return;
+    chatsRef.current.scrollTop = chatsRef.current.scrollHeight;
+  }, [chats]);
+  const id = useId();
+  const [textarea, setTextarea] = useState("");
+  const [expand, setExpand] = useState(false);
+  const handleClickExpand = async () => {
+    setExpand(true);
+    if (chats.length === 0) {
+      pushChat({
+        type: "bot",
+        body: <Wait />,
+      });
+      await sleep(1000);
+      popChat();
+      pushChat({
+        type: "bot",
+        body: <p>こんにちは 👋 探したいキーワードを教えてください</p>,
+      });
+      await sleep(1000);
+      pushChat({
+        type: "bot",
+        body: <Wait />,
+      });
+      await sleep(1000);
+      popChat();
+      pushChat({
+        type: "bot",
+        body: (
+          <p>
+            例えば「育成」とか「指導　人材」のような単語で聞いてみてください
+          </p>
+        ),
+      });
+    }
+  };
+  const handleClickCollapse = () => {
+    setExpand(false);
+  };
+  const handleChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setTextarea(event.target.value);
+  };
+  const handleClickSend = async () => {
+    const keyword = textarea;
+    setTextarea("");
+    pushChat({ type: "user", body: <p>{keyword}</p> });
+    await sleep(1000);
+    pushChat({
+      type: "bot",
+      body: <p>「{keyword}」ですね！探してみます 🚲</p>,
+    });
+    pushChat({
+      type: "bot",
+      body: <Wait />,
+    });
+    await sleep(1000);
```

Comment:
データ取得のための時間を勘案すると待機アニメーションを表示するために1秒は待ちすぎという気がする

```suggestion
    await sleep(500);
```